tests/src/core/testqgsexpression.cpp

```diff
@@ -1218,26 +1218,18 @@ class TestQgsExpression: public QObject
       QTest::newRow( "is_multipart null" ) << "is_multipart(NULL)" << false << QVariant();
       QTest::newRow( "z_max no 3D" ) << "z_max(geom_from_wkt('POINT (0 0)'))" << false << QVariant();
       QTest::newRow( "z_max NULL" ) << "z_max(geom_from_wkt(NULL))" << false << QVariant();
-      QTest::newRow( "z_max point Z NaN" ) << "z_max(geom_from_wkt('PointZ (1 1 nan)'))" << false << QVariant( QVariant::Double );
```

Comment:
Can you restore these tests?

Reply:
Yes, but it will not return a double since NaN in a WKT is a mistake. It will returns a NULL (QVariant::Invalid)